test/Microsoft.Extensions.DependencyInjection.Ordered.Tests/ServiceCollectionOrderedServiceExtensionsTest.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection.Ordered;
+using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public class ServiceCollectionOrderedServiceExtensionsTest
```

Comment:
Shouldn't these tests be added to the SpecificationTests to replace `RegistrationOrderIsPreservedWhenServicesAreIEnumerableResolved`? In theory, containers should get this for "free" once they `Populate` methods are updated to deal with the new factoring of the existing service descriptors while ignoring the ordered descriptors. Still it would be good to verify this.